service/controller/v12/resource/instance/create.go

```diff
@@ -667,3 +695,90 @@ func isNodeDrained(drainerConfigs []corev1alpha1.DrainerConfig, instanceName str
 
 	return false
 }
+
+func (r *Resource) handleDeploymentCompletedStatus(ctx context.Context, customObject providerv1alpha1.AzureConfig) error {
+	deploymentsClient, err := r.getDeploymentsClient(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	d, err := deploymentsClient.Get(ctx, key.ClusterID(customObject), key.VmssDeploymentName)
+	if IsDeploymentNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deployment not found")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for creation")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	s := *d.Properties.ProvisioningState
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment is in state '%s'", s))
+
+	if key.IsSucceededProvisioningState(s) {
+		computedDeployment, err := r.newDeployment(ctx, customObject, nil)
+		if controllercontext.IsInvalidContext(err) {
```

Comment:
Similar to above, lift the error handling to top level so that control flow is also logged & managed there. Masking should be enough at this level.